client/ayon_blender/api/lib.py

```diff
@@ -562,3 +562,32 @@ def collect_animation_defs(create_context, step=True, fps=False):
         defs.append(fps_def)
 
     return defs
+
+
+def get_cache_modifiers(obj, modifier_type="MESH_SEQUENCE_CACHE"):
+    modifiers_dict = {}
+    modifiers = []
+    modifier_names = [modifier.name for modifier in obj.modifiers
+                      if modifier.type == modifier_type]
+    if modifier_names:
+        modifiers = [modifier for modifier in obj.modifiers
+                     if modifier.type == modifier_type]
+        modifiers_dict = {
+            obj.name: (
+                modifier for modifier in obj.modifiers
+                if modifier.type == modifier_type
+            )
+        }
+    else:
+        for sub_obj in obj.children:
+            for ob in sub_obj.children:
+                cache_modifiers = [modifier for modifier in ob.modifiers
+                                   if modifier.type == modifier_type]
+                modifiers.extend(cache_modifiers)
+                modifiers_dict.update({
+                    ob.name: (
+                        modifier for modifier in ob.modifiers
+                        if modifier.type == modifier_type
+                    )
+                })
+    return modifiers_dict
```

Comment:
What's going on here? Why doing so many type checks?
And why do we care about specifically the 'children's children' in the else case?

```suggestion
    modifiers_dict = {}
    modifiers = [modifier for modifier in obj.modifiers
                 if modifier.type == modifier_type]
    if modifiers:
        modifiers_dict[obj.name] = modifiers
    else:
        # QUESTION: Why do we care about the obj's children's children?
        for sub_obj in obj.children:
            for ob in sub_obj.children:
                cache_modifiers = [modifier for modifier in ob.modifiers
                                   if modifier.type == modifier_type]
                modifiers_dict[ob.name] = cache_modifiers
    return modifiers_dict
```

Isn't this the same? but again- why the 'children' iterations? Seems off.

Reply:
I know it doesn't make sense but when I tested with Libor's assets, the modifiers creation lasted inside the geo. The screenshot may explain you more.
![image](https://github.com/user-attachments/assets/07ca65ed-59c8-4daa-acda-5a35ba293ab4)

Reply:
That's correct behavior - but we should just iterate **all** objects, not a specific objects childrens' children?
Also, there's two types of caches, `MeshSequenceCache` for vertex data and there's a constraint or whatever it was I believe that's present on alembic transformations (that are not vertex changes but animated transforms).

Reply:
> That's correct behavior - but we should just iterate **all** objects, not a specific objects childrens' children? Also, there's two types of caches, `MeshSequenceCache` for vertex data and there's a constraint or whatever it was I believe that's present on alembic transformations (that are not vertex changes but animated transforms).

you mean the transform constraint? We used it for the alembic camera loader. It used to be there in the cache loader, but it somehow breaks MeshSequenceCache during implementation of both modifiers, so I removed it.

Reply:
An object _may_ have both, have one of them or none of them. It depends on the animation in the alembic file.

- If the object has no vertex animation (e.g. no skinned deformations) then it will get no alembic mesh modifier
- If the object has no transform animation (e.g. no actual animated transforms) then it will get no constraint modifier.

Let me know if you need me to investigate further

Reply:
> An object _may_ have both, have one of them or none of them. It depends on the animation in the alembic file.
> 
> * If the object has no vertex animation (e.g. no skinned deformations) then it will get no alembic mesh modifier
> * If the object has no transform animation (e.g. no actual animated transforms) then it will get no constraint modifier.
> 
> Let me know if you need me to investigate further

As mentioned above, we can re-implement and we need to set up the condition where we check on the transform and geometry of the asset_group children, and set up the modifier.